biz.aQute.repository/src/aQute/bnd/repository/p2/provider/P2Indexer.java

```diff
@@ -112,6 +135,12 @@ public File apply(File t) {
 		return link;
 	}
 
+	private boolean needsCopy(Path sourcePath, Path destPath) {
+		final File dest = destPath.toFile();
+
+		return !dest.exists() || dest.length() != sourcePath.toFile().length();
```

Comment:
So if we could not create a symlink, we try to shortcut the copy if the files are the same. But I think a length test is pretty weak. We really need a sha match, no?